addons/l10n_ar/wizards/account_debit_note.py

```diff
@@ -0,0 +1,26 @@
+from odoo import models
+
+
+class AccountDebitNote(models.TransientModel):
+
+    _inherit = 'account.debit.note'
+
+    def _prepare_default_values(self, move):
+        """ Properly compute debit type document we need to match to the original document """
+        default_values = super()._prepare_default_values(move)
+        if self.country_code == 'AR' and move.journal_id.l10n_latam_use_documents:
+
+            # Here we are following the same approach or Peru: set the l10n_latam_document_type_id
+            # before the move creation, But in this case we have multiple debit document types, and
+            # we use the  l10n_latam_available_document_type_ids field result in order to set the
+            # proper debit note related to the original move document type.
+            debit_note = self.env['account.move'].new({
```

Comment:
Instead of the new, we could also just do _get_latam_documents_domain on the invoice itself and add the debit_note constraint